src/Chapter4.hs

```diff
@@ -489,6 +514,19 @@ Implement the 'Applicative' instance for our 'List' type.
   type.
 -}
 
+instance Applicative List where
+    pure x = Cons x Empty
+    Empty <*> _        = Empty
+    _     <*> Empty    = Empty
+    (Cons f fs) <*> xs = fmap f xs `addHelper` (fs <*> xs)
+
+
+addHelper :: List a -> List a -> List a
+addHelper Empty       ys = ys
+addHelper (Cons x xs) ys = Cons x (addHelper xs ys)
```

Comment:
Perfect!

You can one more case to make your solution even more efficient:

```haskell
addHelper xs Empty = xs
```

In that case, you won't need to walk through each element of the first list when the second one is empty and return the list instantly 🚤 

One naming suggestion, I would call this function simply `append`. If you want to learn more standard abstractions in Haskell, you can implement the `Semigroup` instance for your `List` type. This typeclass is used to combine values of the same type, and its behaviour for lists is exactly the list append 🙂 

* http://hackage.haskell.org/package/base-4.14.1.0/docs/Data-Semigroup.html#t:Semigroup

Reply:
I implement `Semigroup` and `Monoid` 😁